api_app/api/routes/workspaces.py

```diff
@@ -543,3 +548,67 @@ async def retrieve_user_resource_operations_by_user_resource_id_and_operation_id
 async def retrieve_user_resource_history_by_user_resource_id(user_resource=Depends(get_user_resource_by_id_from_path), user=Depends(get_current_workspace_owner_or_researcher_user_or_airlock_manager), resource_history_repo=Depends(get_repository(ResourceHistoryRepository))) -> ResourceHistoryInList:
     validate_user_has_valid_role_for_user_resource(user, user_resource)
     return ResourceHistoryInList(resource_history=await resource_history_repo.get_resource_history_by_resource_id(resource_id=user_resource.id))
+
+@workspaces_shared_router.get("/workspaces/{workspace_id}/assignable-users", response_model=AssignableUsersInResponse, name=strings.API_GET_ASSIGNABLE_USERS)
+async def get_assignable_users(workspace=Depends(get_workspace_by_id_from_path)) -> AssignableUsersInResponse:
+    if _is_user_management_enabled() is False:
+        logger.exception("Getting Assignable Users failed - User management is disabled.  Enable via the USER_MANAGEMENT_ENABLED environment variable.")
+        raise HTTPException(status_code=status.HTTP_405_METHOD_NOT_ALLOWED, detail=strings.USER_MANAGEMENT_DISABLED)
+
+
+    access_service = get_access_service()
+    assignable_users = access_service.get_assignable_users()
+    return AssignableUsersInResponse(assignable_users=assignable_users)
+
+
+@workspaces_shared_router.get("/workspaces/{workspace_id}/roles", response_model=RolesInResponse, name=strings.API_GET_WORKSPACE_ROLES)
+async def get_workspace_roles(workspace=Depends(get_workspace_by_id_from_path)) -> RolesInResponse:
+    access_service = get_access_service()
+    roles = access_service.get_workspace_roles(workspace)
+    return RolesInResponse(roles=roles)
+
+@workspaces_shared_router.post("/workspaces/{workspace_id}/users/assign", status_code=status.HTTP_202_ACCEPTED, name=strings.API_ASSIGN_WORKSPACE_USER)
```

Comment:
Where is this limited to TRE Admins?

Reply:
Good spot, it is limited in the UI but I need to add it here too.

Reply:
The workspace users router now requires TRE Admin